diff --git a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
index 53d9056..d5500c3 100644
--- a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
+++ b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNCustomerCenterModule.kt
@@ -63,7 +63,7 @@ internal class RNCustomerCenterModule(
     fun presentCustomerCenter(
         promise: Promise
     ) {
-        reactApplicationContext.currentActivity?.let {
+        reactApplicationContext.getCurrentActivity()?.let {
             customerCenterPromise = promise
             presentCustomerCenterFromActivity(it)
         } ?: run {
diff --git a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
index 1684a2b..bc61bec 100644
--- a/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
+++ b/node_modules/react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/RNPaywallsModule.kt
@@ -26,8 +26,9 @@ internal class RNPaywallsModule(
 
     private val currentFragmentActivity: FragmentActivity?
         get() {
-            return when (val currentActivity = reactApplicationContext.currentActivity) {
-                is FragmentActivity -> currentActivity
+            val activity = reactApplicationContext.getCurrentActivity()
+            return when (activity) {
+                is FragmentActivity -> activity
                 else -> {
                     Log.e(NAME, "RevenueCat paywalls require applications to use a FragmentActivity")
                     null
